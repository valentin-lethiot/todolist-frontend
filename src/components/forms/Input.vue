<script setup lang="ts">
type Props = {
  label: string
  modelValue: string
}
defineProps<Props>()

const emit = defineEmits(['update:modelValue'])

const emitNewValue = (event: any) => {
  emit('update:modelValue', event.target.value)
}
</script>

<template>
  <div>
    <label class="block mb-1 text-md font-medium text-white">{{ label }}</label>
    <input
      :value="modelValue"
      type="text"
      class="bg-gray-50 border text-sm rounded-lg block w-full p-2 bg-gray-700 border-gray-600 text-white"
      required
      @input="emitNewValue"
    />
  </div>
</template>
