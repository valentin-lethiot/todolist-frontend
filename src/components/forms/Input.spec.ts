import { beforeEach, expect, it, describe } from 'vitest'
import Input from './Input.vue'
import { mount, shallowMount, VueWrapper } from '@vue/test-utils'

describe('Input', () => {
  let wrapper: VueWrapper

  const defaultProps = {
    label: 'label',
    modelValue: '',
  }

  beforeEach(() => {
    wrapper = mount(Input, {
      props: defaultProps,
    })
  })

  describe('render', () => {
    it('should render the input', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('events', () => {
    it('should emit update:modelValue event when input changes', async () => {
      const input = wrapper.find('input')
      await input.setValue('new value')

      expect(wrapper.emitted('update:modelValue')).toBeTruthy()
      expect(wrapper.emitted('update:modelValue')![0]).toEqual(['new value'])
    })
  })

  describe('props', () => {
    it('should display the label', () => {
      const label = wrapper.find('label')
      expect(label.text()).toBe('label')
    })

    it('should bind the modelValue to input value', () => {
      const input = wrapper.find('input')
      expect(input.element.value).toBe('')

      wrapper = mount(Input, {
        props: {
          ...defaultProps,
          modelValue: 'test value',
        },
      })

      expect(wrapper.find('input').element.value).toBe('test value')
    })
  })
})
