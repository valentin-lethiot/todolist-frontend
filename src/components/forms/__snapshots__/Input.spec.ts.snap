// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > render > should render the input 1`] = `"<div><label class="block mb-1 text-md font-medium text-white">label</label><input type="text" class="bg-gray-50 border text-sm rounded-lg block w-full p-2 bg-gray-700 border-gray-600 text-white" required="" value=""></div>"`;
